ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,13 +358,31 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+      checkUnreadyConnectionsStatus();
       this.metrics.selectorIORate.inc();
     }
     long endIo = time.milliseconds();
     this.metrics.selectorIOTime.update(endIo - endSelect);
   }
 
   /**
+   * Check readiness for unready connections and add to completed list if ready
+   */
+  private void checkUnreadyConnectionsStatus() {
+    Iterator<Map.Entry<String, Long>> iterator = connectionPendingHandshakeTimer.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry<String, Long> entry = iterator.next();
+      if (isChannelReady(entry.getKey())) {
+        connected.add(entry.getKey());
+        metrics.selectorConnectionCreated.inc();
+        metrics.selectorPerceivedSslHandshakeTime.update(System.currentTimeMillis() - entry.getValue());
```

Comment:
Good point. I think this metric can be avoided and that would simplify the code.